beacon_node/network/src/sync/tests/range.rs

```diff
@@ -439,3 +551,65 @@ fn pause_and_resume_on_ee_offline() {
     // The head chain and finalized chain (2) should be in the processing queue
     rig.expect_chain_segments(2);
 }
+
+/// To attempt to finalize the peer's status finalized checkpoint we synced to its finalized epoch +
+/// 2 epochs + 1 slot.
+const EXTRA_SYNCED_EPOCHS: u64 = 2 + 1;
+
+#[test]
+fn finalized_sync_enough_global_custody_peers_few_chain_peers() {
+    // Run for all forks
+    let mut r = TestRig::test_setup();
+    // This test creates enough global custody peers to satisfy column queries but only adds few
+    // peers to the chain
+    r.new_connected_peers_for_peerdas();
+
+    let advanced_epochs: u64 = 2;
+    let remote_info = r.finalized_remote_info_advanced_by(advanced_epochs.into());
+
+    // Current priorization only sends batches to idle peers, so we need enough peers for each batch
+    // TODO: Test this with a single peer in the chain, it should still work
+    r.add_random_peers(
+        remote_info,
+        (advanced_epochs + EXTRA_SYNCED_EPOCHS) as usize,
+    );
+    r.assert_state(RangeSyncType::Finalized);
+
+    let last_epoch = advanced_epochs + EXTRA_SYNCED_EPOCHS;
+    r.complete_and_process_range_sync_until(last_epoch, filter());
+}
+
+#[test]
+fn finalized_sync_not_enough_custody_peers_on_start() {
+    let mut r = TestRig::test_setup();
+    // Only run post-PeerDAS
+    if !r.fork_name.fulu_enabled() {
+        return;
+    }
+
+    let advanced_epochs: u64 = 2;
+    let remote_info = r.finalized_remote_info_advanced_by(advanced_epochs.into());
+
+    // Unikely that the single peer we added has enough columns for us. Find a way to make this test
+    // deterministic.
+    r.add_random_peer_not_supernode(remote_info.clone());
```

Comment:
We could just thread a fixed seed rng to these functions when generating peers?

```
StdRng::seed_from_u64(0xDEADBEEF0BAD5EEDu64)
```

Reply:
Made tests deterministic https://github.com/sigp/lighthouse/pull/6872/commits/2021abcd515bc4ac71c63c6f40e5c026ce18592e